pallets/xcmp-queue/src/lib.rs

```diff
@@ -390,6 +455,20 @@ impl<T: Config> Pallet<T> {
 								remaining_fragments = last_remaining_fragments;
 								break
 							},
+							Err(XcmError::WeightLimitReached(required))
+								if required > max_individual_weight =>
```

Comment:
AFAICT This is only ever triggered if `required > max_weight`, thus `max_individual_weight` is ignored unless we run out of weight for processing XCMs.
The UMP has the two conditions switched: the `required > max_individual_weight` is checked first: 
https://github.com/paritytech/polkadot/pull/3575/files#diff-2893612305e6b9c3b5c48fe27288f735b343e5ce4cca69459fe89f75e282d10bR489